### PR TITLE
[App Service] `az webapp create`: Add `--public-network-access` parameter to support enabling public access

### PR DESCRIPTION
**Pull request by bmis**
**Related command**

az webapp create

**Description**

Adding new argument public-network-access to the az webapp create command-line

**Testing Guide**

This command will enable "Allow public access"
az webapp create --public-network-access Enabled --name app-ardemo-prod-15 --resource-group rg-ardemo --plan asp-ardemo-linux-prod-01 --runtime "PHP:8.0" 

This command will disable "Allow public access"
az webapp create --public-network-access Disabled --name app-ardemo-prod-15 --resource-group rg-ardemo --plan asp-ardemo-linux-prod-01 --runtime "PHP:8.0" 

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -153,6 +154,7 @@ def load_arguments(self, _):
                    local_context_attribute=LocalContextAttribute(name='plan_name', actions=[LocalContextAction.GET]))
         c.argument('vnet', help="Name or resource ID of the regional virtual network. If there are multiple vnets of the same name across different resource groups, use vnet resource id to specify which vnet to use. If vnet name is used, by default, the vnet in the same resource group as the webapp will be used. Must be used with --subnet argument.")
         c.argument('subnet', help="Name or resource ID of the pre-existing subnet to have the webapp join. The --vnet is argument also needed if specifying subnet by name.")
+        c.argument('public_network_access', help="Please define the --public-network-access parameter", arg_type=get_enum_type(PUBLIC_NETWORK_ACCESS_MODES))
```
Comment (StrawnSC) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:157]: The help text here should describe what the parameter does, for instance, `help="Enable or disable public access to the Web App"`.

**Review by StrawnSC — APPROVED** [review]
Just one comment on the help text. Otherwise, looks good to me, thanks!

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py (hunk 1 above)
Comment (bmis) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py]: I changed the description of the help, thanks.

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@StrawnSC Could you please help review this PR again?

**Review by StrawnSC — APPROVED** [review]

**Pull request merged**